Source/ARTChannel.m

```diff
@@ -109,7 +109,7 @@ - (void)publish:(NSString *)name message:(ARTMessage *)message extras:(id<ARTJso
     }
     
     // Checked after encoding, so that the client can receive callback with encoding errors
-    if ([self exceedMaxSize:@[message]]) {
+    if ([self exceedMaxSize:@[messagesWithDataEncoded]]) {
```

Comment:
Please can you provide some context for the change in 5385610a8375b90a45d53a6b151cd216628c0bbd? It is not accompanied by any test changes, and I don't know why you would encode a message's data before calculating its size (other than possibly as an optimisation). If you look at https://sdk.ably.com/builds/ably/specification/main/features/#TM6 you'll see that the data types that it refers to are the user-submitted data types (e.g. there's no way that you'd have an `object` or `array` data property per TM6b if you'd already encoded the data).